### ISSUE TITLE
False positive: const_static_lifetime associated constants

### ISSUE THREAD

**Issue by alexheretic** [rust-lang-nursery/rust-clippy#2456]
Clippy and rustc don't seem to agree when it comes to declaring static associated constant lifetimes.
```rust
trait Foo {
    const BAR: &'static str = "bar";
}
```

clippy
```
warning: Constants have by default a `'static` lifetime
 --> src/main.rs:5:17
  |
5 |     const BAR: &'static str = "bar";
  |                -^^^^^^^---- help: consider removing `'static`: `&str`
```
rustc
```
error[E0106]: missing lifetime specifier
 --> src/main.rs:5:16
  |
5 |     const BAR: &str = "bar";
  |                ^ expected lifetime parameter
```

**Comment by goodmanjonathan** [comment]
This should have been fixed in #2443. Have you tried to update clippy recently?

**Comment by alexheretic** [comment]
Thanks, this indeed does not occur on current master
